src/builder/builder.go

```diff
@@ -141,6 +142,14 @@ func (b *Builder) ReadVersions() {
 	}
 	b.Clearver = strings.TrimSpace(string(ver))
 	b.Clearver = strings.Replace(b.Clearver, "\n", "", -1)
+
+	ver, err = ioutil.ReadFile(b.Versiondir + "/.clearurl")
+	if err != nil {
+		helpers.PrintError(err)
+		os.Exit(1)
+	}
+	b.Upstreamurl = strings.TrimSpace(string(ver))
+	b.Upstreamurl = strings.Replace(b.Upstreamurl, "\n", "", -1)
```

Comment:
I understand this is copying the existing convention used above.  @tmarcu, do you know what edge case this is trying to account for?  The [documentation for strings.TrimSpace](https://golang.org/pkg/strings/#TrimSpace) says that it already includes newlines in its definition of whitespace, so isn't this line redundant?

Reply:
The edge case is where the new line is in the middle of the string since TrimSpace only remove leading and trailing characters.

Reply:
This ^

Reply:
To be honest, I find this surprising. You're expecting .clearversion to potentially contain, for example, "18\n910", but still want to treat that as valid input? Why not skip the strings.TrimSpace and just do a full regexp s/\s// on the entire string? As is, it allows "18 \t   \t   910".  Or, simpler, since clear version is always an integer, why not just do `if _, err := strconv.Atoi(ver)` and whitelist only valid input, rather than parse out invalid input?

Reply:
Agreed. We shouldn't require so much tinkering to parse something so simple. It seems like a better approach to just invalidate anything that is not the plain number.  If some other piece of software is generating this in a weird format, it is probably better to just patch that software instead to output in a proper format.

Reply:
This particular data is an URL, not a number.

Reply:
Haha, yes, `strconv.Atoi(ver)` wouldn't work for you like it would above. Sanitizing/whitelisting will be different for different expected input.

In your particular case, you certainly *could* try to sanitize the input against some kind of "regex for valid URLs", but I would probably err on the side of basic cleanup like `strings.TrimSpace`. It'll error out below if you try to use an invalid URL.